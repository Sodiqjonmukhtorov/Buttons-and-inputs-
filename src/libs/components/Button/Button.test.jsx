// eslint-disable-next-line no-unused-vars
import React from "react";
import renderer from "react-test-renderer";
import { describe, expect, it, test } from "vitest";
import { Button } from "./index";

describe("Buttons", () => {
  it("Buttons component renders correctly", () => {
    const component = renderer.create(<Button name="Click Me" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("The Buttons prop works", () => {
    const component = renderer.create(
      <Button
        onClick={() => {
          console.log("working");
        }}
        variant="primary"
        type="submit"
        style={"md"}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
