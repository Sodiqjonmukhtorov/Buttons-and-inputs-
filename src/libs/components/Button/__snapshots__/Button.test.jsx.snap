// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Buttons > Buttons component renders correctly 1`] = `
<button
  className=""
>
  <span
    className="p-4 font-[Raleway]"
  >
    Click Me
  </span>
</button>
`;

exports[`Buttons > The Buttons prop works 1`] = `
<button
  className="md"
  onClick={[Function]}
  type="submit"
>
  <span
    className="p-4 font-[Raleway]"
  >
    Button
  </span>
</button>
`;
